src/retry/retryUtility.ts

```diff
@@ -110,45 +91,43 @@ export async function apply(
   request.locationRouting.routeToLocation(locationEndpoint);
   try {
     const { result, headers } = await (httpsRequest as Promise<Response<any>>);
-    headers[Constants.ThrottleRetryCount] = resourceThrottleRetryPolicy.currentRetryAttemptCount;
-    headers[Constants.ThrottleRetryWaitTimeInMs] = resourceThrottleRetryPolicy.cummulativeWaitTimeinMilliseconds;
+    headers[Constants.ThrottleRetryCount] = retryPolicies.resourceThrottleRetryPolicy.currentRetryAttemptCount;
+    headers[Constants.ThrottleRetryWaitTimeInMs] =
+      retryPolicies.resourceThrottleRetryPolicy.cummulativeWaitTimeinMilliseconds;
     return { result, headers };
   } catch (err) {
     // TODO: any error
-    let retryPolicy: IRetryPolicy = null;
+    let retryPolicy: RetryPolicy = null;
     const headers = err.headers || {};
     if (err.code === StatusCodes.Forbidden && err.substatus === SubStatusCodes.WriteForbidden) {
-      retryPolicy = endpointDiscoveryRetryPolicy;
+      retryPolicy = retryPolicies.endpointDiscoveryRetryPolicy;
     } else if (err.code === StatusCodes.TooManyRequests) {
-      retryPolicy = resourceThrottleRetryPolicy;
+      retryPolicy = retryPolicies.resourceThrottleRetryPolicy;
     } else if (err.code === StatusCodes.NotFound && err.substatus === SubStatusCodes.ReadSessionNotAvailable) {
-      retryPolicy = sessionReadRetryPolicy;
+      retryPolicy = retryPolicies.sessionReadRetryPolicy;
     } else {
-      retryPolicy = defaultRetryPolicy;
+      retryPolicy = retryPolicies.defaultRetryPolicy;
     }
     const results = await retryPolicy.shouldRetry(err, retryContext);
     if (!results) {
-      headers[Constants.ThrottleRetryCount] = resourceThrottleRetryPolicy.currentRetryAttemptCount;
-      headers[Constants.ThrottleRetryWaitTimeInMs] = resourceThrottleRetryPolicy.cummulativeWaitTimeinMilliseconds;
+      headers[Constants.ThrottleRetryCount] = retryPolicies.resourceThrottleRetryPolicy.currentRetryAttemptCount;
+      headers[Constants.ThrottleRetryWaitTimeInMs] =
+        retryPolicies.resourceThrottleRetryPolicy.cummulativeWaitTimeinMilliseconds;
       err.headers = { ...err.headers, ...headers };
       throw err;
     } else {
       request.retryCount++;
```

Comment:
It looks like newUrl not being used is a bug. I think I broke in the commit after this one: https://github.com/Azure/azure-cosmos-js/blame/b22194817b958aa117205c059b513c9a54f413f3/src/retry/retryUtility.ts#L120-L122